src/ray/gcs/gcs_client/accessor.cc

```diff
@@ -559,11 +559,14 @@ void NodeInfoAccessor::HandleNotification(const GcsNodeInfo &node_info) {
   // Add the notification to our cache.
   RAY_LOG(INFO) << "Received notification for node id = " << node_id
                 << ", IsAlive = " << is_alive;
+
+  auto &node = node_cache_[node_id];
   if (is_alive) {
-    node_cache_[node_id] = node_info;
+    node = node_info;
```

Comment:
Hmm I remember I had some unexpected bug when I had this type of code (assign a value to map referenced obj). Idk if it was similar to this, but why don't we use CopyFrom instead of directly assigning it?

Reply:
Will CopyFrom keep the existing field value if this field is not set in the copy source?

Reply:
I think MergeFrom is better. Modified. Would you please have a look?

Reply:
Is mergeFrom overwrite the existing values? I like the current code (lot simpler), but MergeFrom sounds like it only updates fields that don’t exist.

also is it possible to write a simple ut for this?

Reply:
Oh sad, MergeFrom does override singular files, but it will concatenate repeated fields and merge map fields. I changed it back, do you have better ideas?

Reply:
I think I am just concerned if assigning a value to referenced variable is safe (it might be, but I have some memory I had issue with that style). If you just do node_cache[node_id] = node_info it is fine for me. If you are confident the current code won’t have an issue, that’s also fine, but I am scared cuz it is not unit tested..

Reply:
I think there won't be any problem if you don't modify the container during using the reference. Do you remember the code when you met the issue?

For the UT, I'll add one.